src/api/rpc_api/author.rs

```diff
@@ -301,243 +298,13 @@ where
 			self.submit_and_watch_opaque_extrinsic_until(encoded_extrinsic, xt_status)?;
 
 		let block_hash = report.block_hash.ok_or(Error::BlockHashNotFound)?;
-		let extrinsic_index =
-			self.retrieve_extrinsic_index_from_block(block_hash, report.extrinsic_hash)?;
-		let block_events = self.fetch_events_from_block(block_hash)?;
-		let extrinsic_events = self.filter_extrinsic_events(block_events, extrinsic_index)?;
+		let extrinsic_events =
+			self.fetch_events_for_extrinsic(block_hash, report.extrinsic_hash)?;
+		// Ensure that the extrins has been successful. If not, return an error.
+		for event in extrinsic_events.clone() {
```

Comment:
Can't the events also be checked by using a reference? (in order to avoid the `clone()`)

Reply:
Ah yes, you're absolutely right. Fixed!